lib/policy/policyValidator.js

```diff
@@ -0,0 +1,106 @@
+'use strict'; // eslint-disable-line strict
+
+const Ajv = require('ajv');
+const userPolicySchema = require('./userPolicySchema');
+const errors = require('../errors');
+
+const ajValidate = new Ajv({ allErrors: true });
+// compiles schema to functions and caches them for all cases
+const userPolicyValidate = ajValidate.compile(userPolicySchema);
+
+// parse ajv errors and build list of erros
+function _parseErrors(ajvErrors) {
+    let parsedErr;
+    ajvErrors.some(err => {
+        const resource = err.dataPath.replace('.', '');
+        if (err.keyword === 'required' && err.params) {
+            const field = err.params.missingProperty;
+            if (field === 'Version') {
+                parsedErr = errors.MissingPolicyVersion;
+            } else if (field === 'Statement') {
+                parsedErr = errors.MissingPolicyStatement;
+            } else if (field === 'Action') {
+                parsedErr = errors.MissingPolicyAction;
+            } else if (field === 'Effect') {
+                parsedErr = errors.MissingPolicyEffect;
+            } else if (field === 'Resource') {
+                parsedErr = errors.MissingPolicyResource;
+            } else {
+                parsedErr = errors.InvalidPolicyDocument;
+            }
+        } else if (err.keyword === 'minItems' && resource === 'Statement') {
+            parsedErr = errors.InvalidPolicyStatement;
+        } else if (err.keyword === 'pattern') {
+            parsedErr = errors.InvalidPolicyDocument;
+        } else if (err.keyword === 'type') {
+            // skip if it's Statement as it does not have enough
+            // error context
+            if (resource === 'Version') {
+                parsedErr = errors.PolicyInvalidVersion;
+            }
+        } else {
+            parsedErr = errors.InvalidPolicyDocument;
+        }
+        return parsedErr instanceof Error;
```

Comment:
The main function returns either `undefined` or `Error`. The test function for Array.some always returns `bool` which would only exit the Array.some not the main function.